### DISCUSSION TITLE
Unmounting screen component

### DISCUSSION THREAD

**Discussion by lukaszradziak** [expo/router#857] [General]
How to handle unmounting screen component in `<Stack>` / `<Stack.Screen>`? 
This code:
```
export default function HomeScreen() {
  useEffect(() => {
    console.log('screen in');
    return () => {
      console.log('screen OUT');
    }
  }, []);
  // ...
}
```

working in `<Slot>` but in `<Stack>` "screen OUT" not appeared

**Comment by redkean** [comment]
use useFocusEffect() hook instead of useEffect
